### DISCUSSION TITLE
Disable automatic following of redirect

### DISCUSSION THREAD

**Discussion by weili-jiang** [baresip/re#477] [Ideas/Proposals]
Currently all incoming 3xx responses are handled internally and followed through. I would like to be able to have some control over whether redirection (3xx) is automatically followed or not, based on what is in the 3xx message.

Would you accept a PR for a new API for `re_sipsess.h` as follows:

```c
int sipsess_follow_redirects(struct sipsess *sess, bool enable);
```

This would set a `bool` flag on the session such that in `invite_resp_handler` can chose not to follow the redirect and just terminate the call with the 3xx message available to the client. I would call it immediately after `sipsess_connect`. Existing clients that do not call this new function will behave as before.

**Comment by sreimers** [comment]
A PR is welcome.
